sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/reflect/DoFnSignatures.java

```diff
@@ -505,6 +514,58 @@ private static void verifyUnsplittableMethods(ErrorReporter errors, DoFnSignatur
             formatType(paramT),
             formatType(expectedOutputReceiverT));
         extraParameters.add(DoFnSignature.Parameter.outputReceiver());
+      } else if (Timer.class.equals(rawType)) {
```

Comment:
It seems like the code in this block could be extracted into a helper method pretty easily?